cmd/bridge/config.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+)
+
+// Config is the top-level console configuration.
+type Config struct {
+	APIVersion    string `yaml:"apiVersion"`
+	Kind          string `yaml:"kind"`
+	ServingInfo   `yaml:"servingInfo"`
+	ClusterInfo   `yaml:"clusterInfo"`
+	Auth          `yaml:"auth"`
+	Customization `yaml:"customization"`
+}
+
+// ServingInfo holds configuration for serving HTTP.
+type ServingInfo struct {
+	BindAddress string `yaml:"bindAddress"`
+	CertFile    string `yaml:"certFile"`
+	KeyFile     string `yaml:"keyFile"`
+
+	// These fields are defined in `HTTPServingInfo`, but are not supported for console. Fail if any are specified.
+	// https://github.com/openshift/api/blob/0cb4131a7636e1ada6b2769edc9118f0fe6844c8/config/v1/types.go#L7-L38
+	BindNetwork           string        `yaml:"bindNetwork"`
+	ClientCA              string        `yaml:"clientCA"`
+	NamedCertificates     []interface{} `yaml:"namedCertificates"`
+	MinTLSVersion         string        `yaml:"minTLSVersion"`
+	CipherSuites          []string      `yaml:"cipherSuites"`
+	MaxRequestsInFlight   int64         `yaml:"maxRequestsInFlight"`
+	RequestTimeoutSeconds int64         `yaml:"requestTimeoutSeconds"`
+}
+
+// ClusterInfo holds information the about the cluster such as master public URL and console public URL.
+type ClusterInfo struct {
+	ConsoleBaseAddress string `yaml:"consoleBaseAddress"`
+	ConsoleBasePath    string `yaml:"consoleBasePath"`
+	MasterPublicURL    string `yaml:"masterPublicURL"`
+}
+
+// Auth holds configuration for authenticating with OpenShift. The auth method is assumed to be "openshift".
+type Auth struct {
+	ClientID         string `yaml:"clientID"`
+	ClientSecretFile string `yaml:"clientSecretFile"`
+	CAFile           string `yaml:"caFile"`
```

Comment:
I thought we wanted something like `oauthEndpointCAFile` to be specific.

Reply:
I don't mind the more specific name if you prefer.